torchmetrics/classification/cosine_similarity.py

```diff
@@ -0,0 +1,91 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Callable, Optional
+
+import torch
+from torch import Tensor, tensor
+
+from torchmetrics.functional.classification.cosine_similarity import (
+    _cosine_similarity_compute,
+    _cosine_similarity_update,
+)
+from torchmetrics.metric import Metric
+
+
+class CosineSimilarity(Metric):
+    r"""
+       Computes the `Cosine Similarity <https://en.wikipedia.org/wiki/Cosine_similarity>`_
+        between targets and predictions:
+       Accepts all input types listed in :ref:`references/modules:input types`.
```

Comment:
This is not correct as it refers to the input types for classification problems.
Instead is should be something similar to 
https://github.com/PyTorchLightning/metrics/blob/652d3f804cad08660842fad440cdead2e986059d/torchmetrics/regression/pearson.py#L36-L39